frontend/src/ProfilePage/Graphs/CompBoxPlot.jsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useRef } from 'react';
```

Comment:
at first glance, the CompBoxPlot and TimeChart graphs have a similar implementation approach where you do some computations and then construct a svg. How much of this code is re-usable? Could we refactor portions like the svg creation?

Reply:
Thanks for the explanation, that makes sense